convert.c

```diff
@@ -256,6 +257,117 @@ static int will_convert_lf_to_crlf(size_t len, struct text_stat *stats,
 
 }
 
+static struct encoding {
+	const char *name;
+	iconv_t *to_git;
+	iconv_t *to_worktree;
+	struct encoding *next;
+} *encoding, **encoding_tail;
+static const char *default_encoding = "utf-8";
+
+static int encode_to_git(const char *path, const char *src, size_t src_len,
+			 struct strbuf *buf, struct encoding *enc)
+{
+#ifndef NO_ICONV
+	char *dst, *re_src;
+	int dst_len, re_src_len;
+
+	if (!enc || (src && !src_len))
+		return 0;
+
+	/*
+	 * Looks like we got called from "would_convert_to_git()".
+	 * Since we have an encoding we must assume that we would encode
+	 * the content.
+	 */
+	if (!buf && !src) {
+		return 1;
+	}
+
+	if (!enc->to_git || !enc->to_worktree) {
+		enc->to_git = xcalloc(1, sizeof(iconv_t));
+		enc->to_git = iconv_open(default_encoding, encoding->name);
+		enc->to_worktree = xcalloc(1, sizeof(iconv_t));
+		enc->to_worktree = iconv_open(encoding->name, default_encoding);
+
+		if (enc->to_git == (iconv_t)-1 ||
+		    enc->to_worktree == (iconv_t)-1)
+			error(_("unsupported encoding %s"), encoding->name);
+	}
+
+	/* Do nothing if the encoding is not supported */
+	if (enc->to_git == (iconv_t)-1 ||
```

Comment:
This should be `*enc->to_git`.

Reply:
Okay, the additional check for null pointers might be unnecessary, because this is checked above.